app/views/stories/feeds/show.json.jbuilder

```diff
@@ -24,6 +24,11 @@ json.array!(@stories) do |article|
   end
 
   json.tag_list article.cached_tag_list_array
-  json.top_comments article.top_comments
   json.extract! article, *article_methods_to_include
+
+  json.top_comments article.top_comments do |comment|
```

Comment:
you could `article.top_comments.select(:id,  ... various fields ..., :user_id)` here, to only load what you need from the DB

Reply:
It's good idea, but it only works for the fields on `comment`: I can't selectively include fields from the `users` that hang off of each comment as part of single query (at least that I could find). I did play around with doing `joins(:user).select(:username, :name)`, but it cancels out the includes and I'm back N+1 land for loading users :/ I even tried some Arel (gasp!), but couldn't find a way to punch through the top level includes (`top_comments`) to the lower level includes (`users`).

I've left in the optimized select of comment fields, but I'm reluctant to delve further given the limited amount of records I expect to be pulling back.

Reply:
thanks for the explanation!